persist/fs/write.go

```diff
@@ -54,25 +64,55 @@ type writer struct {
 	err        error
 }
 
+type indexEntry struct {
```

Comment:
Is there any benefit in optimizing the size of this struct?

From what I can tell, offsetForSummary never stores a value that is different than offset, so you could probably replace it with a bool. That doesn't actually change the size of the struct, but if you then replace size (or offset but size seems safer) with uint32 and re-order the fields, you can save 8 bytes per struct making it 40 bytes instead of 48 (if this is meaningful at all):

https://play.golang.org/p/dXGawT5YxL

Might be premature, I don't know how hot this is

Reply:
Although it might make your sort faster too just by virtue of being able to fit more of them in the cache lines

Reply:
The two offset values are actually different.  One is the offset into the data file, the `offsetForSummary` is the offset of this record in the index file that the summary entry will point to.  I'll update these to be `dataOffset` and `indexOffset` perhaps.

Yeah size could be reduced to be a uint32 for sure, offsets probably shouldn't be in case we ever have a file >4gb.  

I'm pretty sure that the size of the struct won't benefit from just a single field being a uint32 though due to alignment.

Reply:
https://github.com/dominikh/go-tools/tree/master/cmd/structlayout might be helpful if you wanna go down this road

Reply:
This one is optimized, in future we can always look at this - agreed.